Added Azure.Automation.ManagedIdentity rule

ArmaanMcleod opened the pull request:
## PR Summary

Fix #1074

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

ArmaanMcleod commented on docs/en/rules/Azure.Automation.ManagedIdentity.md:
```diff
@@ -0,0 +1,97 @@
+severity: Important
+pillar: Security
+category: Identity and access management
+resource: Automation Service
```
Wondering if we should use **Automation Account** for resource?

BernieWhite replied:
@ArmaanMcleod Yes, probably a good idea to change this.

BernieWhite reviewed (approved):
@ArmaanMcleod Nice work! All good to merge. Simple cases might be good to do in YAML if they can be.

ArmaanMcleod commented:
> @ArmaanMcleod Nice work! All good to merge. Simple cases might be good to do in YAML if they can be.

@BernieWhite Thanks 🙂. I renamed resource to Automation Account and converted rule to YAML. 

Looks like during rule generation the `Azure.Template.ExpressionLength` rule docs also got updated. Let me know if this looks ok to you.

BernieWhite reviewed (approved):
@ArmaanMcleod Looks good to me.
